src/hooks/useNFT.ts

```diff
@@ -2,22 +2,30 @@ import { abi, bytecode } from "@/lib/nft";
 import { useMutation } from "@tanstack/react-query";
 import { toast } from "sonner";
 import { useAccount, usePublicClient, useWalletClient } from "wagmi";
+import { useAddressWhitelist } from "@/hooks/useAddressWhitelist";
 
 export const useNFT = () => {
   const walletClient = useWalletClient();
   const publicClient = usePublicClient();
   const { address } = useAccount();
-  const DEFAULT_BASE_URI = "/images/nft/logo.png";
+  const { data: whitelist } = useAddressWhitelist();
 
   const deploy = useMutation({
     mutationFn: async ({ name, symbol }: { name: string; symbol: string }) => {
       if (!walletClient?.data) throw new Error("Wallet client not found");
       if (!publicClient) throw new Error("Public client not found");
+      if (!whitelist?.nftImageURI) throw new Error("NFT image not found");
+
+      const nftImage = {
+        name: `${name} NFT Image`,
+        description: `${name} project NFT Image`,
+        image: `ipfs://${whitelist?.nftImageURI}`,
+      };
```

Comment:
Do you know if it worked? 
I was thinking about
```suggestion
      const nftImage = whitelist?.nftImageURI || DEFAULT_BASE_URI;
```